File: app-mobile/app/src/main/java/com/example/appmobile/models/Peca.java
package com.example.appmobile.models;

public class Peca {
    private int id;
    private String nome;
    private double preco;
    private String tipo;
    private boolean selecionada;

    public Peca(int id, String nome, double preco, String tipo) {
        this.id = id;
        this.nome = nome;
        this.preco = preco;
        this.tipo = tipo;
        this.selecionada = false;
    }

    // Getters e Setters
    public int getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public double getPreco() {
        return preco;
    }

    public String getTipo() {
        return tipo;
    }

    public boolean isSelecionada() {
        return selecionada;
    }

    public void setSelecionada(boolean selecionada) {
        this.selecionada = selecionada;
    }

    @Override
    public String toString() {
        return nome + " (R$ " + String.format("%.2f", preco) + ")";
    }
}

File: app-mobile/app/src/main/java/com/example/appmobile/models/Peca.kt
package com.example.appmobile.models

data class Peca(
    val id: Int,
    val nome: String,
    val preco: Double,
    val tipo: String,
    val ganho_hp: Int = 0,
    val descricao: String = ""
) {
    override fun toString(): String {
        return "$nome (R$ ${String.format("%.2f", preco)})"
    }
}
